pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/FilterContext.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.service.plugin;
+
+import lombok.Data;
+import org.apache.bookkeeper.mledger.ManagedCursor;
+import org.apache.pulsar.broker.service.EntryBatchIndexesAcks;
+import org.apache.pulsar.broker.service.EntryBatchSizes;
+import org.apache.pulsar.broker.service.SendMessageInfo;
+import org.apache.pulsar.broker.service.Subscription;
+import org.apache.pulsar.broker.service.SubscriptionOption;
+import org.apache.pulsar.common.api.proto.MessageMetadata;
+
+@Data
+public class FilterContext {
+    private EntryBatchSizes batchSizes;
+    private SendMessageInfo sendMessageInfo;
+    private EntryBatchIndexesAcks indexesAcks;
+    private ManagedCursor cursor;
+    private boolean isReplayRead;
+    private Subscription subscription;
+    private SubscriptionOption subscriptionOption;
+    private MessageMetadata msgMetadata;
```

Comment:
Do we need to expose all of them to the filter? I don't think the plugin need to filter the messages by batchSizes, sendMessageInfo, indexesAcks, cursor, isReplayRead?

And if we expose Subscription, SubscriptionOption, MessageMetadata, users will get the write access, maybe we'd better to only expose the topic name, the subname and subType and a readonly MessageMetadata or a copy is more reasonable here? For the filter, it is always in the read-only mode.